iceoryx_utils/source/posix_wrapper/periodic_timer.cpp

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2021 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/posix_wrapper/periodic_timer.hpp"
+
+using namespace iox::units::duration_literals;
+
+namespace iox
+{
+namespace posix
+{
+PeriodicTimer::PeriodicTimer(const iox::units::Duration interval, const iox::units::Duration delayThreshold) noexcept
+    : m_interval(interval)
+    , m_delayThreshold(delayThreshold)
+{
+    m_waitSemaphore =
+        std::move(posix::Semaphore::create(posix::CreateUnnamedSharedMemorySemaphore, 0U)
+                      .or_else([](posix::SemaphoreError&) {
+                          errorHandler(Error::kROUDI_APP__FAILED_TO_CREATE_SEMAPHORE, nullptr, ErrorLevel::FATAL);
```

Comment:
according to the error-handling.md the error handler shall not be used in utils/hoofs. Please use `cxx::Ensures(false && "Could not create Semaphore for PeriodicTimer!")`

Reply:
Okay. Replaced with the cxx::Ensures.